### PR TITLE
Update genrules.cpp

### PR DESCRIPTION
**Pull request by Elfinhope**
Updated genrules.cpp to reflect change from War/Trade/Magic faction types to Martial/Magic.  Included some minor grammatical changes as well.  Changes also made to Combat section rules for movement after battle.  This addresses issues #102 and #106.

### CODE REVIEW

File: genrules.cpp [hunk 1]
```diff
@@ -171,21 +171,21 @@ void Game::WriteFactionTypeDescription(std::ostringstream& buffer, Faction &fac)
...
-			if (Globals->FACTION_ACTIVITY == FactionActivityRules::MARTIAL_MERGED) { 
+			if (Globals->FACTION_ACTIVITY == FactionActivityRules::MARTIAL_MERGED) {
 				buffer << "perform tax and trade in " << nma << " " << plural(nma, "region", "regions");
 			}
 		}
 
 		count++;
 	}
-
+/*
```
Comment (artyomtrityak) [inline genrules.cpp:181]: why did you commented this part?

File: genrules.cpp [hunk 2]
```diff
@@ -3064,14 +3082,20 @@ int Game::GenRules(const AString &rules, const AString &css,
...
+			"directly, but when completed are placed in Fleets. "
+			"Fleets may contain one or more ships, and "
 			"may be entered like other buildings.";
 		f.Paragraph(temp);
 		temp = "";
 		if (Globals->FACTION_LIMIT_TYPE == GameDefs::FACLIM_FACTION_TYPES) {
-			temp += "Only factions with at least one faction point spent on "
-				"trade can issue ";
+			if (Globals->BUILD_NO_TRADE) {
+				temp = +"Any faction ";
```
Comment (gelzis) [inline genrules.cpp:3092]: I guess this was intended as `temp += "Any faction ";`

File: genrules.cpp (hunk 1 above)
Comment (Elfinhope) [reply genrules.cpp:181]: I commented that out a couple of days ago...When it generated the rules it was printing "as well as 5 apprentices" twice. Once with the information about mages and a second time at the end of the description block.  Commenting out this code got rid of the second time.  I left it in but commented out as I was not sure what effect that change would have under different rules configurations.  It works properly with current configuration, but haven't had time to make sure that the output is correct with Martial (not Martial Merged) and original War/Trade/Magic rules.

File: genrules.cpp (hunk 2 above)
Comment (Elfinhope) [reply genrules.cpp:3092]: Yes not sure how that compiled that way. I'm pretty sure I checked the output and it was correct... I'll double check it tonight when I get off work.

Comment (Elfinhope) [reply genrules.cpp]: This has been addressed and patch pushed.

**Comment by artyomtrityak** [comment]
@Elfinhope is it ready to review / merge?

**Review by valdisz — APPROVED** [review]

**Pull request closed without merging**